tests/tests/wallet_tests.cpp

```diff
@@ -70,10 +71,80 @@ BOOST_FIXTURE_TEST_SUITE(wallet_tests, database_fixture)
           for (auto info : derived_keys) {
               string description = (string) info.pub_key;
-              BOOST_CHECK_EQUAL(0, description.find(expected_prefix));
+              BOOST_CHECK_EQUAL(0u, description.find(expected_prefix));
           }
 
       } FC_LOG_AND_RETHROW()
   }
 
+  BOOST_AUTO_TEST_CASE(verify_account_authority) {
+      try {
+         
+         ACTORS( (nathan) );
+         graphene::app::database_api db_api(db);
+
+         // good keys
+         flat_set<public_key_type> public_keys;
+         public_keys.emplace(nathan_public_key);
+         BOOST_CHECK(db_api.verify_account_authority( "nathan", public_keys));
+
+         // bad keys
+         flat_set<public_key_type> bad_public_keys;
+         bad_public_keys.emplace(public_key_type("BTS6MkMxwBjFWmcDjXRoJ4mW9Hd4LCSPwtv9tKG1qYW5Kgu4AhoZy"));
+         BOOST_CHECK(!db_api.verify_account_authority( "nathan", bad_public_keys));
+
+      } FC_LOG_AND_RETHROW()
+  }
+
+BOOST_AUTO_TEST_CASE( any_two_of_three )
+{
+   try {
+      fc::ecc::private_key nathan_key1 = fc::ecc::private_key::regenerate(fc::digest("key1"));
+      fc::ecc::private_key nathan_key2 = fc::ecc::private_key::regenerate(fc::digest("key2"));
+      fc::ecc::private_key nathan_key3 = fc::ecc::private_key::regenerate(fc::digest("key3"));
+      const account_object& nathan = create_account("nathan", nathan_key1.get_public_key() );
+      fund(nathan);
+      graphene::app::database_api db_api(db);
+
+      try {
+         account_update_operation op;
+         op.account = nathan.id;
+         op.active = authority(2, public_key_type(nathan_key1.get_public_key()), 1, 
+               public_key_type(nathan_key2.get_public_key()), 1, public_key_type(nathan_key3.get_public_key()), 1);
+         op.owner = *op.active;
+         trx.operations.push_back(op);
+         sign(trx, nathan_key1);
+         PUSH_TX( db, trx, database::skip_transaction_dupe_check );
+         trx.clear();
+      } FC_CAPTURE_AND_RETHROW ((nathan.active))
+
+      // two keys should work
+      {
+      	flat_set<public_key_type> public_keys;
+      	public_keys.emplace(nathan_key1.get_public_key());
+      	public_keys.emplace(nathan_key2.get_public_key());
+      	BOOST_CHECK(db_api.verify_account_authority("nathan", public_keys));
+      }
+
+      // the other two keys should work
+      {
+     	   flat_set<public_key_type> public_keys;
+      	public_keys.emplace(nathan_key2.get_public_key());
+      	public_keys.emplace(nathan_key3.get_public_key());
+     	   BOOST_CHECK(db_api.verify_account_authority("nathan", public_keys));
+      }
+
+      // just one key should not work
+      {
+     	   flat_set<public_key_type> public_keys;
+         public_keys.emplace(nathan_key1.get_public_key());
+     	   BOOST_CHECK(!db_api.verify_account_authority("nathan", public_keys));
+      }
+   } catch (fc::exception& e) {
+      edump((e.to_detail_string()));
+      throw;
+   }
+}
```

Comment:
Please add a test where nathan's account has a multisig from one key and two other accounts.

Reply:
Added what I think you are asking. It is not much different than the last, so I may be misunderstanding your request.

Reply:
IMHO we should check different combinations as in `any_two_of_three` test, perhaps the same combinations, but with 3 accounts, 1 key per 1 account, just my thoughts ...